packages/@aws-cdk/integ-runner/lib/runner/integ-test-runner.ts

```diff
@@ -135,28 +139,24 @@ export class IntegTestRunner extends IntegRunner {
         `You need to manually checkout the snapshot directory ${this.snapshotDir}` +
         'from the merge-base (https://git-scm.com/docs/git-merge-base)',
       );
-      logger.warning('error: %s', e);
+      logger.warning('error: %s', formatError(e));
     }
 
     // if we found the base branch then get the merge-base (most recent common commit)
     // and checkout the snapshot using that commit
     if (baseBranch) {
-      const relativeSnapshotDir = path.relative(this.directory, this.snapshotDir);
+      const relativeSnapshotDir = path.relative(gitCwd, this.snapshotDir);
 
+      const checkoutCommand = [...git, 'checkout', [...git, 'merge-base', 'HEAD', baseBranch], '--', relativeSnapshotDir];
       try {
-        const base = exec(['git', 'merge-base', 'HEAD', baseBranch], {
-          cwd,
-        });
-        exec(['git', 'checkout', base, '--', relativeSnapshotDir], {
-          cwd,
-        });
+        execWithSubShell(checkoutCommand);
```

Comment:
Wouldn't doing this:

```ts
exec(renderCommand(checkoutCommand), { shell: true })
```

Achieve the same result, without needing the `execWithSubShell` function? Might fail on windows though, just wondering if you considered/tested that.

Reply:
I considered but didn't want to change the way this is executed, precisely because of Windows. This really is just a more condensed way of writing the same code we had before in away that can also be rendered as a string.

Reply:
Also `shell: true` tends to be flagged up from a security perspective. It's not really an issue for client side tooling, but I'd like to avoid if we can.